client/state/selectors/is-woo-jpc-flow.ts

```diff
@@ -11,10 +11,10 @@ const isLegacyJetpackWooOnboardingFlow = ( state: AppState ) => {
 	return 'woocommerce-onboarding' === get( getCurrentQueryArguments( state ), 'from' );
 };
 
-const isWooCommercePaymentsOnboardingFlow = ( state: AppState ) => {
+export const isWooCommercePaymentsOnboardingFlow = ( state: AppState ) => {
 	const from =
 		get( getInitialQueryArguments( state ), 'from' ) === 'woocommerce-payments' ||
-		get( getCurrentQueryArguments( state ), 'from' ) === 'woocommerce-payments';
+		get( getCurrentQueryArguments( state ), 'from' ) === 'woocommerce-onboarding';
```

Comment:
I wonder if we need to check both the `getInitialQueryArguments` and `getCurrentQueryArguments` for both `woocommerce-payments` and `woocommerce-onboarding`? 

Also, do you think there's any danger this will also disable the progress bar in unintended cases (like a WooCommerce Core profiler onboarding)? I ask because I'm unfamiliar with where this `woocommerce-onboarding` argument is passed.

Reply:
> I wonder if we need to check both the getInitialQueryArguments and getCurrentQueryArguments for both woocommerce-payments and woocommerce-onboarding?

Oh, that's a good call. I'll work on it!

> Also, do you think there's any danger this will also disable the progress bar in unintended cases (like a WooCommerce Core profiler onboarding)? I ask because I'm unfamiliar with where this woocommerce-onboarding argument is passed.

Yes, that's what I was trying to do with the introduction of new query arguments, but we want to limit the number of new parameters.

I'm not sure if this change will cause other issues. @vladolaru, do you have anyone in mind we can ask? 

Alternatively, we can consider all other cases and manually test them to ensure they're not breaking.

Reply:
> I wonder if we need to check both the getInitialQueryArguments and getCurrentQueryArguments for both woocommerce-payments and woocommerce-onboarding?

Good call, Dan. We definitely need to do that.

> Also, do you think there's any danger this will also disable the progress bar in unintended cases (like a WooCommerce Core profiler onboarding)? I ask because I'm unfamiliar with where this woocommerce-onboarding argument is passed.

I see no danger in doing that since we need both the `from` and the `plugin` to be just right (see `from && plugin` below).

Reply:
@dmallory42 I added the changes here https://github.com/Automattic/wp-calypso/pull/102714/commits/17fb869ce6de2dd04bf9d655ec50c6e31c45a6ad